cap/modules/deposit/serializers/schemas/json.py

```diff
@@ -139,3 +174,55 @@ def can_user_review(self, obj):
             can_review = ReviewDepositPermission(obj['deposit']).can()
             if can_review:
                 return True
+
+    def get_schema_permission_info(self, obj):
+        name, version = (
+            obj['deposit'].schema.name,
+            obj['deposit'].schema.version,
+        )
+        permission_field = obj['deposit'].schema.config.get('x-cap-permission')
+
+        x_cap_fields = dict()
+        if permission_field:
+            schema = current_jsonschemas.get_schema(
+                obj['deposit'].schema.deposit_path,
+                with_refs=True,
+                resolved=True,
+            )
+            x_cap_fields = parse_schema_permission_info(name, version, schema)
+
+        return x_cap_fields
+
+    def can_user_edit_field(self, perm_obj):
```

Comment:
why duplicate logic in `can_user_edit_field` and `get_user_schema_permission_info`?

Reply:
This was for showing options on how we can keep the permission info.